### ISSUE TITLE
Crash on world load containing sorting machine

### ISSUE THREAD

**Issue by Wupp** [elytra/Teckle#26]
Hi,

I've had this issue where every time I would place a sorting machine into my world and reload that world, the game would crash. I would be able to load back into the world, but my sorting machines would reset and upon quitting the world and reloading, the crash would occur again. I have my sorting machines set up in this manner: https://i.imgur.com/SCMo1Gw.png. This doesn't occur when the machines are set up so one machine into several different colours, but i would like a large variety of items in one chest at once.

I am playing on Minecraft 1.12.2 and using Teckle verion 1.12-5.394. Here is an example of a crash log due to this bug: https://gist.github.com/Wupp/5a6b49962b28e35483b135bb3218ca1f

**Comment by darkevilmac** [comment]
I need a bit more info if I'm going to go reproduce this. Show the inventory of each sorting machine so I can copy to the best of my abilities. Not sure what those posts are by the way, if it's a redstone clock they're not needed as the sorting machine has one built in.

**Comment by Wupp** [comment]
Here's a video where I set up the system causing the crash: https://www.youtube.com/watch?v=I8GrhzrjxEw&feature=youtu.be . The machine at the beginning was there to demonstrate that in that particular setup, the machine would not crash, and reload to prove this. I build the machine, and let it function for a short period of time, then reload the world, which crashes the game. When I relaunch, the world loads, but the sorting machines inventory has been cleared. I then attempt to relaunch without further touching the sorting machine, which crashes the game again. After relaunching for the last time, i show that the first machine is not affected, as its inventory is not cleared.

BTW thanks for the tip about the automated timer in the machine, I had no clue!

**Comment by darkevilmac** [comment]
Thanks for that video it's a huge help, I'll take a look at this now.

**Comment by darkevilmac** [comment]
I just went ahead and remade the structure you have in the video but couldn't get the same result.

Replace your local teckle jar with the one hosted [here](https://unascribed.com/f/2397e581_Teckle-1.12-5.409-dirty.jar) and give me the full log file once you experience the crash. It has a small debug message that I can use to try and figure this out.

**Comment by Wupp** [comment]
I tried replacing the jar and the mod wasn't identified by the game. When I tried opening it with WinRAR, it wouldn't open and suggested "the archive is either in unknown format or damaged". For reference the release version opened fine.

**Comment by exaskye** [comment]
Yeah apparently Firefox decided to pause the download without telling me and just made an empty file, because "people don't usually download this file".

Try this one: https://unascribed.com/f/e9d57b9c_Teckle-1.12-5.409-dirty.jar

**Comment by Wupp** [comment]
Here you go: https://gist.github.com/Wupp/72be141a4ecef2f0cad9b7d9b3e4a40a

**Comment by exaskye** [comment]
We'll need the full log, not just the crash report.

**Comment by Wupp** [comment]
Oh, sorry about that, I should've read more thoroughly. I believe this is what you're looking for though: https://gist.github.com/Wupp/0afa8cbcdeb8864f1cdc69622c28de4e

**Comment by exaskye** [comment]
Yep, that contains the debug log darkevilmac was talking about. :+1:

**Comment by darkevilmac** [comment]
This bug is really weird, the world isnt null and there's no way the filter id or buffer id is null. Whipping up a new build with even more debug data.

**Comment by darkevilmac** [comment]
@Wupp use [this](https://matrix.cybre.space/_matrix/media/v1/download/cybre.space/rSKIzdVkcQzrakehmobKmTEm).

**Comment by Wupp** [comment]
Interestingly enough, I've replicated the same setup from the previous versions to produce the bug, and it hasn't crashed. I'll continue to use this version to test, so if it starts crashing again and I can figure out the cause, I'll post back here. Otherwise, I have no clue as to how to reproduce the bug in this debug version.

**Comment by darkevilmac** [comment]
This is just weird as that build is identical short of some log calls.

**Comment by wormzjl** [comment]
Somehow I got the same crash (using the version on jenkins -.-)

**Comment by adrwa** [comment]
Is this issue still happening or has it been magically fixed?
